### PR TITLE
ESLint: enable rule: @typescript-eslint/no-shadow

### PR DESCRIPTION
**Pull request by alfonsomthd**
* Avoid variable shadowing to prevent potential bugs.
* Address issues found.

### CODE REVIEW

File: packages/shared/yaml-editor/YAMLEditor.tsx [hunk 1]
```diff
@@ -92,10 +92,10 @@ const YAMLEditor: React.FC<YAMLEditorProps> = ({
     setOpen(false);
   };
 
-  const updateYAML = (obj, model, newNamespace, newName) => {
+  const updateYAML = (obj, newModel, newNamespace, newName) => {
```
Comment (SanjalKatiyar) [inline packages/shared/yaml-editor/YAMLEditor.tsx:95]: I can understand the intention, but this name is little misleading... there is no "newModel", "model" object is not getting updated...

Comment (alfonsomthd) [reply packages/shared/yaml-editor/YAMLEditor.tsx:95]: I'll update the name to `k8sModel`.

File: packages/mco/components/mco-dashboard/data-policy/cluster-app-card/common.tsx [hunk 2]
```diff
@@ -252,16 +252,19 @@ const AppDropdown: React.FC<Partial<ClusterAppDropdownProps>> = ({
     _event: React.MouseEvent<Element, MouseEvent> | undefined,
     itemId: string
   ) => {
-    const [namespace, name] = getNSAndNameFromId(itemId);
-    setAppSet({ namespace, name });
+    const [ItemNamespace, ItemName] = getNSAndNameFromId(itemId);
+    setAppSet({ namespace: ItemNamespace, name: ItemName });
```
[the remark this answers is not in the document]
Comment (alfonsomthd) [reply packages/mco/components/mco-dashboard/data-policy/cluster-app-card/common.tsx:256]: Done.

**Pull request merged**